### ISSUE TITLE
Horizontal orientantion scroll not working

### ISSUE THREAD

**Issue by tsmrecki** [ch4vi/FlowLayout#3]
Hey,
I've implemented your layout manager for my RecyclerView which is supposed to have horizontal scrolling. The items are drawn and the flow layout works, but I can't make the items scroll horizontally. The flag for orientantion is set to RecyclerView.HORIZONTAL. 
Is there a fix to that?

**Comment by ch4vi** [comment]
Did you try the sample app in HORIZONTAL orientation? The sample has both orientations implemented and I just tried and they are working and scrolling. With no more information I can't see  where the problem could be.

**Comment by tsmrecki** [comment]
I use Java so I didn't cp the code from the sample app, but here's my code:

`FlowLayoutManager flowLayoutManager = new FlowLayoutManager(3, RecyclerView.HORIZONTAL, new FlowLayoutManager.Interface() {
            @NotNull
            @Override
            public Pair<Integer, Integer> getProportionalSizeForChild(int position) {
                return new Pair<>(1, 1);
            }


        });

        latestPhotosRV.setHasFixedSize(true);
        latestPhotosRV.addItemDecoration(flowLayoutManager.new InsetDecoration(getActivity()));
        latestPhotosRV.setLayoutManager(flowLayoutManager);`

The only difference is that my RecyclerView is in fragment, not in activity.
Adapter is just plain extension of RV.Adapter.
I checked the FlowLayoutManager canScrollHorizontally flag, and it's set to false if that's of any help to you.

**Comment by ch4vi** [comment]
I just tried to use a Recycler in a Fragment with Java and it works... With horizontal and vertical scrolling, the difference that I can see is `flowLayoutManager.new InsetDecoration(getActivity())` I'm using `getContext()` instead of `getActivity()`

```
class Adapter extends RecyclerView.Adapter<Adapter.ViewHolder> {
    private String[] data;

    Adapter(String[] data) {
        this.data = data;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.card_view, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public int getItemCount() {
        return data.length;
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        holder.text.setText(data[position]);
    }

    class ViewHolder extends RecyclerView.ViewHolder {
        protected TextView text;

        ViewHolder(View view) {
            super(view);
            text = (TextView) view.findViewById(R.id.text_view);
        }
    }
}
```

```
public class MyFragment extends Fragment {

    public MyFragment() {
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_fragment, container, false);
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);


        RecyclerView recyclerView = (RecyclerView) view.findViewById(R.id.recycler_view);
        recyclerView.setHasFixedSize(true);

        recyclerView.getItemAnimator().setAddDuration(1000);
        recyclerView.getItemAnimator().setChangeDuration(1000);
        recyclerView.getItemAnimator().setMoveDuration(1000);
        recyclerView.getItemAnimator().setRemoveDuration(1000);

        FlowLayoutManager manager = new FlowLayoutManager(3, RecyclerView.HORIZONTAL, new FlowLayoutManager.Interface() {
            @NotNull
            @Override
            public Pair<Integer, Integer> getProportionalSizeForChild(int position) {
                switch (position) {
                    case 0:
                        return new Pair<>(1, 1);
                    case 1:
                        return new Pair<>(2, 2);
                    case 2:
                        return new Pair<>(4, 1);
                    case 3:
                        return new Pair<>(3, 2);
                    default:
                        return new Pair<>(1, 1);
                }
            }
        });
        recyclerView.setLayoutManager(manager);
        FlowLayoutManager.InsetDecoration inset = manager.new InsetDecoration(getContext());
        recyclerView.addItemDecoration(inset);

        recyclerView.setAdapter(new Adapter(new String[]{"0", "1", "2", "3", "4", "5"}));

    }
}
```

**Comment by tsmrecki** [comment]
Hey, I finally figured out the problem.
If you set the height of the RecyclerView to _match_parent_, then the scroll is working.
But if you set its height (for example, _240dp_ as I did), the scroll is not working.

`   <android.support.v7.widget.RecyclerView
        android:layout_width="match_parent"
        android:layout_height="240dp"
        android:id="@+id/recycler_view"/>`

Could you please check this issue?

**Comment by ch4vi** [comment]
You were right! It was a silly mistake. 
I'll upload a new version ASAP.

Thank you.
fix 9b0f5c6
